src/parser/lexer.mll

```diff
@@ -70,6 +70,32 @@ let trim_trailing_blank_lines : string -> string = fun s ->
     in
     String.sub s 0 trim_from
 
+let trim_leading_whitespace : string -> string = fun s ->
+  let count_leading_whitespace : string -> int = fun line ->
+    let rec count_leading_whitespace' : int -> int = fun index ->
+      if index >= String.length line then
+        index
+      else
+        match line.[index] with
+        | ' ' | '\t' -> count_leading_whitespace' (index + 1)
+        | _ -> index
+    in
+    count_leading_whitespace' 0
+  in
+  let lines = Str.(split (regexp "\n") s) in
+  let least_amount_of_whitespace =
+    lines
+    |> List.map count_leading_whitespace
+    |> List.fold_left min max_int
```

Comment:
I'd add a comment, that using `max_int` is okay because if the list is empty, we will not use the result of the `fold` (in the `map` below).